tests/test_serializer_nested.py

```diff
@@ -38,3 +38,88 @@ def test_nested_serialize_empty(self):
         }
         serializer = self.Serializer()
         assert serializer.data == expected_data
+
+
+class TestNestedSerializerWithMany:
+    def setup(self):
+        class PaymentInfoSerializer(serializers.Serializer):
+            url = serializers.URLField()
+            amount = serializers.DecimalField(max_digits=6, decimal_places=2)
+
+        class NestedSerializer(serializers.Serializer):
+            foo = serializers.CharField()
+
+        class AcceptRequestSerializer(serializers.Serializer):
+            ERROR_MESSAGE = '`payment_info` is required under this condition'
+
+            nested = NestedSerializer(many=True)
+            payment_info = PaymentInfoSerializer(many=True, allow_null=True)
+
+            def validate_payment_info(self, value):
+                if value is None and not self.context['condition_to_allow_null']:
+                    raise serializers.ValidationError(self.ERROR_MESSAGE)
+
+                return value
+
+        self.Serializer = AcceptRequestSerializer
+
+    def get_serializer(self, data, condition_to_allow_null=True):
+        return self.Serializer(
+            data=data,
+            context={'condition_to_allow_null': condition_to_allow_null})
+
+    def test_null_allowed_if_allow_null_is_set(self):
+        input_data = {
+            'nested': [{'foo': 'bar'}],
+            'payment_info': None
+        }
+        expected_data = {
+            'nested': [{'foo': 'bar'}],
+            'payment_info': None
+        }
+        serializer = self.get_serializer(input_data)
+
+        assert serializer.is_valid(), serializer.errors
+        assert serializer.validated_data == expected_data
+
+    def test_null_is_not_allowed_if_allow_null_is_not_set(self):
+        input_data = {
+            'nested': None,
+            'payment_info': None
+        }
+        serializer = self.get_serializer(input_data)
+
+        assert not serializer.is_valid()
+        assert set(serializer.errors) == set(['nested'])
+        assert serializer.errors['nested'][0] == serializer.error_messages['null']
```

Comment:
Can we just do `expected_errors = ...` and then `assert serializer.errors == expected_errors` for simplicity?